bindings/rust/s2n-tls/src/testing/s2n_tls.rs

```diff
@@ -383,4 +384,65 @@ mod tests {
         assert_eq!(callback.count(), 1);
         Ok(())
     }
+
+    #[test]
+    fn trust_location() -> Result<(), Error> {
+        struct AcceptAllHostnames {}
+        impl VerifyHostNameCallback for AcceptAllHostnames {
+            fn verify_host_name(&self, _: &str) -> bool {
+                true
+            }
+        }
+
+        let pem_dir = Path::new(concat!(env!("CARGO_MANIFEST_DIR"), "/../../../tests/pems"));
+        let mut cert = pem_dir.to_path_buf();
+        cert.push("rsa_4096_sha512_client_cert.pem");
+        let mut key = pem_dir.to_path_buf();
+        key.push("rsa_4096_sha512_client_key.pem");
+
+        let mut builder = crate::config::Builder::new();
+        builder.set_security_policy(&security::DEFAULT_TLS13)?;
+        builder.set_verify_host_callback(AcceptAllHostnames {})?;
+        builder.load_pem(&fs::read(&cert)?, &fs::read(&key)?)?;
+        builder.trust_location(Some(&cert), None)?;
```

Comment:
Unintuitively, "s2n_tls_pair" actually performs a handshake. If the trust location wasn't set, the handshake would fail because the certificate would be untrusted. If the handshake fails, "s2n_tls_pair" panics.